### PR TITLE
2982-feat: Clear SelectableList reset scoped state

### LINKED ISSUE #2982: Clear SelectableList ScopedStates
After closing the selectableList, scoped states should be reset back to default value.

So scoped state hook is required.

### PR DESCRIPTION
**Pull request by Kanav-Arora**
Closes #2982

### CODE REVIEW

File: packages/twenty-front/src/modules/command-menu/hooks/useCommandMenu.ts [hunk 1]
```diff
@@ -14,6 +15,9 @@ export const useCommandMenu = () => {
   const navigate = useNavigate();
   const setIsCommandMenuOpened = useSetRecoilState(isCommandMenuOpenedState);
   const setCommands = useSetRecoilState(commandMenuCommandsState);
+  const { resetSelectableListStates } = useSelectableList({
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/command-menu/hooks/useCommandMenu.ts:18]: I think we should not talk about "states" in the API of selectableList. It feels to me that states are an internal logic here.
What about `resetSelectedItem` instead?

**Review by charlesBochet — COMMENTED** [review]
@Kanav-Arora Thank you! I have merged main into your branch and I have left comments

**Comment by Kanav-Arora** [comment]
Hi @charlesBochet 
Done

File: packages/twenty-front/src/modules/ui/layout/selectable-list/hooks/useSelectableList.ts [hunk 2]
```diff
@@ -30,11 +31,32 @@ export const useSelectableList = (props?: UseSelectableListProps) => {
...
+    resetIsSelectedItemIdSelector();
+  };
+
   return {
     setSelectableItemIds,
     isSelectedItemId,
     setSelectableListOnEnter,
     selectableListId: scopeId,
     isSelectedItemIdSelector,
+    resetSelectedItemIdState,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/selectable-list/hooks/useSelectableList.ts:56]: we should not expose these as we don't want it to be part of the external API of useSelectableList, 
we should only expose resetSelectedItem

**Review by charlesBochet — APPROVED** [review]
Left a last comment :)

**Comment by Kanav-Arora** [comment]
@charlesBochet Done

**Comment by charlesBochet** [comment]
Thank you!

**Pull request merged**